### PR TITLE
Upgrade biome

### PR DESCRIPTION
**Pull request by oscarvz**
# Description

Upgrade Biome & add linting exclusions where necessary.
Changes work well with the [Biome (nightly)](https://marketplace.visualstudio.com/items?itemName=biomejs.biome-nightly) VS Code extension, as the current stable `1.6.2` version doesn't support the latest changes.

# Checklist

~~- [ ] The changes have been tested to be backwards compatible.~~
~~- [ ] The OpenAPI schema and generated client have been updated.~~
~~- [ ] New models module has been added to api generator xtask~~
~~- [ ] New types/fields are [well-documented and annotated](/CONTRIBUTING.md#adding-types-and-their-annotations).~~
~~- [ ] The CHANGELOG is updated.~~

> Please link any related PRs in other repositories that depend on this:

After merging, please merge related PRs ASAP, so others don't get blocked.

### CODE REVIEW

File: fiberplane-charts/src/MetricsChart/MetricsChart.tsx [hunk 1]
```diff
@@ -101,7 +101,9 @@ const InnerMetricsChart = memo(function InnerMetricsChart(
 
   // When the timeseries changes, we want to reset the id so that the legend
   // is re-rendered. This resets the size values for all legend items
-  // biome-ignore lint/nursery/useExhaustiveDependencies: this is intentional
+  /*
+    biome-ignore lint/correctness/useExhaustiveDependencies: this is intentional
+  */
```
Comment (arendjr) [inline fiberplane-charts/src/MetricsChart/MetricsChart.tsx]: Why did this become a multiline comment? Seems a bit overkill.

File: fiberplane-charts/src/Mondrian/generator/utils/createMetricBuckets.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import type { Buckets } from "../types";
 export function createMetricBuckets<T>(
   timeseriesData: Array<Timeseries>,
   reducer: (current: T | undefined, metricValue: number) => T,
+  // biome-ignore lint/suspicious/noConfusingVoidType: this is intentional
```
Comment (arendjr) [inline fiberplane-charts/src/Mondrian/generator/utils/createMetricBuckets.ts]:
```suggestion
  // biome-ignore lint/suspicious/noConfusingVoidType: without this, the overload doesn't match
```

File: fiberplane-charts/src/hooks/useHandler.ts [hunk 3]
```diff
@@ -2,6 +2,10 @@ import { useCallback, useRef } from "react";
 
 const noDeps: Array<void> = [];
 
+/*
+  biome-ignore lint/complexity/noBannedTypes: we can't know what the argument &
+  return types are
+*/
```
Comment (arendjr) [inline fiberplane-charts/src/hooks/useHandler.ts]:
```suggestion
// biome-ignore lint/complexity/noBannedTypes: must match arbitrary functions
```

File: fiberplane-charts/src/MetricsChart/MetricsChart.tsx (hunk 1 above)
Comment (oscarvz) [reply fiberplane-charts/src/MetricsChart/MetricsChart.tsx]: I'm quite strict in following the 80 characters per line 😊

**Review by arendjr — APPROVED** [review]
:rocket:

**Review by stephlow — APPROVED** [review]

**Pull request merged**